### PR TITLE
LPD - 17467 Importing Object - Custom State Flow doesn't resemble JSON file

### PR DESCRIPTION
**Pull request by victorKammerer**
Solution code and tests for [LPD-17467](https://liferay.atlassian.net/browse/LPD-17467).

### CODE REVIEW

**Comment by victorKammerer** [comment]
Waiting [LPD-19577](https://liferay.atlassian.net/browse/LPD-19577) to finally adapt the implementation here and then merge.

**Comment by rijema** [comment]
ci:test:sf

**Comment by victorKammerer** [comment]
ci:test:relevant

File: modules/apps/object/object-test/src/testIntegration/resources/com/liferay/object/web/internal/object/definitions/portlet/action/test/dependencies/test-object-definition-state-manager.json [hunk 1]
```diff
@@ -0,0 +1,419 @@
...
+				}
+			],
+			"readOnly": "false",
+			"readOnlyConditionExpression": "",
+			"required": true,
+			"state": true,
+			"system": false,
+			"type": "String",
+			"unique": false
+		}
```
Comment (MarinhoFeliphe) [inline modules/apps/object/object-test/src/testIntegration/resources/com/liferay/object/web/internal/object/definitions/portlet/action/test/dependencies/test-object-definition-state-manager.json:220]: Hey @victorKammerer can we just put this part of the **json** inside of the `test-object-definition.json` used [here](https://github.com/liferay-objects/liferay-portal/blob/b3fe5897be00934775f11dc84627d39c9b18a6a1/modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/web/internal/object/definitions/portlet/action/test/ObjectDefinitionExportImportTest.java#L49)?

**Review by MarinhoFeliphe — CHANGES_REQUESTED** [review]
Hey @victorKammerer please see my comments.

**Review by MarinhoFeliphe — CHANGES_REQUESTED** [review]
Hey @victorKammerer please see my comments.

**Comment by victorKammerer** [comment]
@MarinhoFeliphe, done!

**Review by MarinhoFeliphe — CHANGES_REQUESTED** [review]
@victorKammerer see my comments,

nice job! :tada:

File: modules/apps/object/object-test/src/testIntegration/resources/com/liferay/object/web/internal/object/definitions/portlet/action/test/dependencies/test-object-definition.json [hunk 2]
```diff
@@ -198,8 +198,106 @@
...
+									},
+									{
+										"key": "underReview"
+									}
+								],
+								"key": "onHold"
+							}
+						]
+					}
+				},
```
Comment (MarinhoFeliphe) [inline modules/apps/object/object-test/src/testIntegration/resources/com/liferay/object/web/internal/object/definitions/portlet/action/test/dependencies/test-object-definition.json:283]: Hey @victorKammerer sort the transitions this way:

````json
{
	"name": "stateFlow",
	"value": {
		"objectStates": [
			{
				"key": "approved",
				"objectStateTransitions": [
					{
						"key": "underReview"
					}
				]
			},
			{
				"key": "denied",
				"objectStateTransitions": [
					{
						"key": "underReview"
					}
				]
			},
			{
				"key": "onHold",
				"objectStateTransitions": [
					{
						"key": "open"
					},
					{
						"key": "underReview"
					}
				]
			},
			{
				"key": "open",
				"objectStateTransitions": [
					{
						"key": "underReview"
					},
					{
						"key": "withdrawn"
					}
				]
			},
			{
				"key": "underReview",
				"objectStateTransitions": [
					{
						"key": "approved"
					},
					{
						"key": "denied"
					},
					{
						"key": "onHold"
					},
					{
						"key": "withdrawn"
					}
				]
			},
			{
				"key": "withdrawn",
				"objectStateTransitions": [
					{
						"key": "open"
					}
				]
			}
		]
	}
}
````

File: modules/apps/object/object-test/src/testIntegration/resources/com/liferay/object/web/internal/object/definitions/portlet/action/test/dependencies/test-object-definition.json [hunk 3]
```diff
@@ -198,8 +198,106 @@
...
+		},
+		{
+			"DBType": "String",
+			"businessType": "Picklist",
+			"externalReferenceCode": "TESTOBJECTFIELD3",
+			"indexed": true,
+			"indexedAsKeyword": false,
+			"indexedLanguageId": "en_US",
+			"label": {
+				"en_US": "testObjectImport"
```
Comment (MarinhoFeliphe) [inline modules/apps/object/object-test/src/testIntegration/resources/com/liferay/object/web/internal/object/definitions/portlet/action/test/dependencies/test-object-definition.json:210]: @victorKammerer the label should be `Test Object Field 3`

File: modules/apps/object/object-test/src/testIntegration/resources/com/liferay/object/web/internal/object/definitions/portlet/action/test/dependencies/test-object-definition.json [hunk 4]
```diff
@@ -198,8 +198,106 @@
...
+					}
+				},
+				{
+					"name": "defaultValueType",
+					"value": "inputAsValue"
+				},
+				{
+					"name": "defaultValue",
+					"value": "open"
+				}
```
Comment (MarinhoFeliphe) [inline modules/apps/object/object-test/src/testIntegration/resources/com/liferay/object/web/internal/object/definitions/portlet/action/test/dependencies/test-object-definition.json:291]: @victorKammerer sort the objectFieldSettings, for example:

````json
				{
					"name": "defaultValue",
					"value": "open"
				},
				{
					"name": "defaultValueType",
					"value": "inputAsValue"
				},
				{
					"name": "stateFlow",
					...
````

**Review by MarinhoFeliphe — CHANGES_REQUESTED** [review]
Hey @victorKammerer I found other SF stuff, could you please take a look and resend

Thanks!

**Comment by victorKammerer** [comment]
All requested changes done,  @MarinhoFeliphe!

**Comment by victorKammerer** [comment]
ci:test:relevant

**Review by MarinhoFeliphe — APPROVED** [review]
LGTM!! :tada:

@victorKammerer  Nice job!

**Comment by MarinhoFeliphe** [comment]
ci:forward

**Comment by MarinhoFeliphe** [comment]
ci:test:object

**Pull request closed without merging**